### PR TITLE
improvement: error message for invalid 'content-type' in subgraph response

### LINKED ISSUE #5359: Error message for invalid 'content-type' in subgraph response doesn't provide enough information
**Describe the bug**
At the moment error message for invalid 'content-type' in subgraph response looks like this:
```
HTTP fetch failed from 'test': subgraph didn't return JSON (expected content-type: application/json or content-type: application/graphql-response+json; found content-type: text/html)
```

It doesn't contain invalid values, and it doesn't have any explanation of what is wrong with a content type.

### PR DESCRIPTION
**Pull request by IvanGoncharov**
Fixes #5359

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [x] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [ ] Integration Tests
    - [x] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Review by Geal — APPROVED** [review]

File: apollo-router/src/services/subgraph_service.rs [hunk 1]
```diff
@@ -1294,37 +1294,45 @@ enum ContentType {
...
-                    service: service_name.to_string(),
-                    reason: format!("subgraph didn't return JSON (expected content-type: {} or content-type: {}; found content-type: {content_type})", APPLICATION_JSON.essence_str(), GRAPHQL_JSON_RESPONSE_HEADER_VALUE),
-                })
             }
+            Some(mime) => Err(format!(
+                "subgraph response contains unsupported content-type: {}",
+                mime,
+            )),
+            None => Err(format!(
+                "subgraph response contains invalid 'content-type' value {:?}",
```
Comment (bnjjj) [inline apollo-router/src/services/subgraph_service.rs:1319]:
```suggestion
                "subgraph response contains invalid 'content-type' header value {:?}",
```

File: apollo-router/src/services/subgraph_service.rs [hunk 2]
```diff
@@ -1698,8 +1706,37 @@ mod tests {
...
+                .body(r#"TEST"#.into())
+                .unwrap())
+        }
+
+        let make_svc = make_service_fn(|_conn| async { Ok::<_, Infallible>(service_fn(handle)) });
+        let server = Server::from_tcp(listener).unwrap().serve(make_svc);
+        server.await.unwrap();
+    }
+
+    // starts a local server emulating a subgraph returning response with missing content_type
```
Comment (bnjjj) [inline apollo-router/src/services/subgraph_service.rs:1723]: I think this comment is wrong

File: .changesets/fix_fix_content_type_error.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+### Improve error message for invalid `content-type` in subgraph response ([Issue #5359](https://github.com/apollographql/router/issues/5359))
+
+To improve their understandability and debuggability the router now includes more details in error messages caused by invalid `content-type` in subgraph response.
+
```
Comment (abernix) [inline .changesets/fix_fix_content_type_error.md:4]:
````suggestion
### Improve error message produced when subgraphs responses don't include an expected `content-type` header value ([Issue #5359](https://github.com/apollographql/router/issues/5359))

To improve a common debuggability challenge when a subgraph response doesn't contain an expected `content-type` header value, the error message produced will include additional details about the error.

Examples:

   * ```
     HTTP fetch failed from 'test': subgraph response contains invalid 'content-type' header value \"application/json,application/json\"; expected content-type: application/json or content-type: application/graphql-response+json
      ```
   * ```
     HTTP fetch failed from 'test': subgraph response does not contain 'content-type' header; expected content-type: application/json or content-type: application/graphql-response+json
      ```
````

**Review by abernix — CHANGES_REQUESTED** [review]
My review only includes a requested edit for the changelog.

**Review by bnjjj — APPROVED** [review]

**Pull request merged**